src/render/direct3d/SDL_render_d3d.c

```diff
@@ -1627,7 +1633,13 @@ D3D_CreateRenderer(SDL_Window *window, Uint32 flags)
     if (window_flags & SDL_WINDOW_FULLSCREEN && (window_flags & SDL_WINDOW_FULLSCREEN_DESKTOP) != SDL_WINDOW_FULLSCREEN_DESKTOP) {
         pparams.Windowed = FALSE;
         pparams.BackBufferFormat = PixelFormatToD3DFMT(fullscreen_mode.format);
-        pparams.FullScreen_RefreshRateInHz = fullscreen_mode.refresh_rate;
+        {
+            double rr = 0;
+            if (fullscreen_mode.refresh_rate.denominator) {
+                rr = (double) fullscreen_mode.refresh_rate.numerator / (double) fullscreen_mode.refresh_rate.denominator;
+            }
+            pparams.FullScreen_RefreshRateInHz = (int) rr;
```

Comment:
You should use SDL_ceil() explicitly here rather than relying on the current rounding mode.

Reply:
```
 302             data->pparams.BackBufferFormat = PixelFormatToD3DFMT(fullscreen_mode.format);
 303             {
 304                 double rr = 0;
 305                 if (fullscreen_mode.refresh_rate.denominator) {
 306                     rr = (double) fullscreen_mode.refresh_rate.numerator / (double) fullscreen_mode.refresh_rate.denominator;
 307                 }
 308                 data->pparams.FullScreen_RefreshRateInHz = (int) SDL_ceil(rr);
 309             }

```